db/migrate/20180109102659_add_unique_indexes_to_containers_tables.rb

```diff
@@ -0,0 +1,122 @@
+              :unique => true,
+              :name   => "index_taggings_unique_multi_column"
+    add_index :computer_systems,
+              %i(managed_entity_id managed_entity_type),
+              :unique => true,
+              :name   => "index_computer_systems_unique_multi_column"
+    add_index :container_env_vars,
+              %i(container_id name value field_path),
+              :unique => true,
+              :name   => "index_container_env_vars_unique_multi_column"
+    add_index :container_limit_items,
+              %i(container_limit_id resource item_type),
+              :unique => true,
+              :name   => "index_container_limit_items_unique_multi_column"
+    add_index :container_port_configs,
+              %i(container_id ems_ref),
+              :unique => true,
+              :name   => "index_container_port_configs_unique_multi_column"
+    add_index :container_quota_items,
+              %i(container_quota_id resource quota_desired quota_enforced quota_observed),
+              :where  => "deleted_on IS NULL",
+              :unique => true
+    add_index :container_quota_scopes,
+              %i(container_quota_id scope),
+              :unique => true
+    add_index :container_service_port_configs,
+              %i(container_service_id name),
+              :unique => true,
+              :name   => "index_container_service_port_configs_unique_multi_column"
+    add_index :container_template_parameters,
+              %i(container_template_id name),
+              :unique => true,
+              :name   => "index_container_template_parameters_unique_multi_column"
+    add_index :container_volumes,
+              %i(parent_id parent_type ems_ref name),
+              :unique => true,
+              :name   => "index_container_volumes_unique_multi_column"
+    add_index :custom_attributes,
+              %i(resource_id resource_type name unique_name section source),
+              :unique => true,
+              :name   => "index_custom_attributes_parameters_unique_multi_column"
+
+    remove_index :hardwares, :vm_or_template_id
+    add_index :hardwares,
+              %i(vm_or_template_id),
+              :where  => "host_id IS NULL AND computer_system_id IS NULL",
+              :unique => true,
+              :name   => "index_hardwares_on_vm_or_template_id"
+
+    remove_index :hardwares, :host_id
+    add_index :hardwares,
+              %i(host_id),
+              :where  => "vm_or_template_id IS NULL AND computer_system_id IS NULL",
+              :unique => true,
+              :name   => "index_hardwares_on_host_id"
+
+    remove_index :hardwares, :computer_system_id
+    add_index :hardwares,
+              %i(computer_system_id),
+              :where  => "vm_or_template_id IS NULL AND host_id IS NULL",
+              :unique => true,
+              :name   => "index_hardwares_on_computer_system_id_"
+
+    remove_index :operating_systems, :vm_or_template_id
```

Comment:
@cben so, this is how we will have to model unique indexes, when table are shared and have nullable parts
(we should redesign the DB though, having N:M mapping tables for vm_or_template_id, host_id and computer_system_id, then those tables can have correct NOT NULL constraints and unique indexes  )

Then for other cases, where index is several columns that are nullable, I think we will have to build the unique :ems_ref out of them